pkg/agent/multicast/mcast_controller.go

```diff
@@ -509,14 +622,57 @@ func (c *Controller) updateQueryGroup() error {
 		memberPorts = append(memberPorts, uint32(iface.OFPort))
 	}
 	// Install OpenFlow group for a new multicast group which has local Pod receivers joined.
-	if err := c.ofClient.InstallMulticastGroup(c.queryGroupId, memberPorts); err != nil {
+	if err := c.ofClient.InstallMulticastGroup(c.queryGroupId, memberPorts, nil); err != nil {
 		return err
 	}
 	klog.V(2).InfoS("Installed OpenFlow group for local receivers", "group", types.McastAllHosts.String(),
 		"ofGroup", c.queryGroupId, "localReceivers", memberPorts)
 	return nil
 }
 
+// syncLocalGroupsToOtherNodes sends IGMP join message to other Nodes in the same cluster to notify what multicast groups
+// are joined by this Node. This function is used only with encap mode.
+func (c *Controller) syncLocalGroupsToOtherNodes() {
+	if c.installedLocalGroups.Len() == 0 {
+		return
+	}
+	localGroups := make([]net.IP, c.installedLocalGroups.Len())
+	c.installedLocalGroupsMutex.RLock()
+	for i := range c.installedLocalGroups.List() {
+		localGroups[i] = net.ParseIP(c.installedLocalGroups.List()[i])
+	}
+	c.installedLocalGroupsMutex.RUnlock()
+	if err := c.igmpSnooper.sendIGMPJoinReport(localGroups); err != nil {
+		klog.ErrorS(err, "Failed to sync local multicast groups to other Nodes")
+	}
+}
+
+func (c *Controller) nodeChanged(e interface{}) {
+	nodes, ok := e.(map[string]*utilip.DualStackIPs)
+	if !ok {
+		klog.InfoS("Failed to cast nodeUpdateEvent")
+		return
+	}
+	var nodeIPs []net.IP
+	updatedNodes := sets.NewString()
+	for _, ips := range nodes {
+		if ips != nil && ips.IPv4 != nil {
+			nodeIPs = append(nodeIPs, ips.IPv4)
+			updatedNodes.Insert(ips.IPv4.String())
+		}
+	}
+	if c.installedNodes.Equal(updatedNodes) {
+		klog.V(2).InfoS("Nodes in the cluster are not changed, ignore the event")
+		return
+	}
+	if err := c.ofClient.InstallMulticastGroup(c.nodeGroupID, nil, nodeIPs); err != nil {
+		klog.ErrorS(err, "Failed to update OpenFlow group for remote Nodes")
+	}
```

Comment:
It doesn't return error and still updates `c.installedNodes`, which could cause data inconsitency and subsequent updates can skip it by mistake.